### DISCUSSION TITLE
Project typeCheckingLevel as `strict`, individual files downgrade to `basic`

### DISCUSSION THREAD

**Discussion by the-witch-king** [microsoft/pyright#4715] [Q&A] [answered]
Hey y'all,

Project i'm working on uses pyright. We use `typeCheckingMode = 'basic'` by default right now. Further, a bunch of our files are in `none` mode, as they're quite old and need to be fixed up first.

However, we want to make `strict` the default for all _new_ files. Our thinking was, in the `pyproject.toml` set `strict` as the default, and add `# pyright: basic` to all existing files that don't already have an explicit mode set.

However, this doesn't seem to work. It appears the `strict` global setting supercedes the per-file `basic` setting. Is there some way to accomplish this?

**Comment by erictraut** [comment, marked as answer]
The recommended way to do this is to use "basic" type checking by default. For new source files or for files that have been "upgraded" to strict typing, include a `# pyright: strict` comment or add them individually to the "strict" configuration list. When entire source directories have been upgraded to strict mode, you can add the directory to the "strict" configuration list and delete any `# pyright: strict` from the corresponding files.

This is the technique I've used to gradually type a large Python source base and eventually upgrade it to strict typing.

**Reply by erictraut** [reply]
Keep in mind that you can also disable specific diagnostic rules within a file by using something like `# pyright: strict, reportPropertyTypeMismatch=none, reportUnusedVariable=none`.
